Content.Server/Storage/Components/StorageFillComponent.cs

```diff
@@ -51,10 +52,20 @@ void IMapInit.MapInit()
 
                 for (var i = 0; i < storageItem.Amount; i++)
                 {
-                    storage.Insert(Owner.EntityManager.SpawnEntity(storageItem.PrototypeId, Owner.Transform.Coordinates));
+                    storage.Insert(
+                        Owner.EntityManager.SpawnEntity(storageItem.PrototypeId, Owner.Transform.Coordinates));
                 }
+
                 if (!string.IsNullOrEmpty(storageItem.GroupId)) alreadySpawnedGroups.Add(storageItem.GroupId);
             }
+
+            // Finished Inserting all
+            // disregard poor quality of code
+            if (Owner.TryGetComponent(out ServerStorageComponent? container))
+            {
+                Owner.EntityManager.EventBus.RaiseEvent(EventSource.Network,
+                    new AfterStorageFillEvent(container.Owner, container.StoredEntities));
+            }
```

Comment:
i don't see why this trygetcomp needs to exist, a storagecomponent is kind of guaranteed if storagefill has gotten this far, and you can just use Owner for the first parameter to AfterStorageFillEvent. as for the second parameter (stored entities uid), it'd be better if you just created a list before the loop from lines 53-60 and added the uids in there. pseudocode like this:

```
var uids = new List<EntityUid>()
for each (item in storageItem)
{
    entity = SpawnEntity(prototype, coords)
    uids.Add(entity.Uid)
    storage.Insert(entity)
}
```

then later just pass the uid list in to AfterStorageFillEvent (you would have to change the event's ContainedEntities field to `IReadOnlyList<EntityUid>`, though

Reply:
also this should be raised local, not network

Reply:
Well, it's there because StorageFill is used to fill more than just storage and we don't care about those at the moment.

Reply:
There's no way this is getting merged as a network event so it needs a change.